File: sample-wear/src/main/java/com/github/droibit/messenger/sample/MainActivity.java
package com.github.droibit.messenger.sample;

import android.app.Activity;
import android.os.Bundle;

import com.droibit.messengerapp.R;
import com.github.droibit.messenger.Messenger;
import com.github.droibit.messenger.sample.model.ConfirmMessageReceiver;
import com.github.droibit.messenger.sample.model.ResponseMessageReceiver;
import com.github.droibit.messenger.sample.model.StandardMessageReceiver;
import com.google.android.gms.common.api.GoogleApiClient;
import com.google.android.gms.wearable.Wearable;

import static com.github.droibit.messenger.sample.model.ConfirmMessageReceiver.*;


public class MainActivity extends Activity implements GoogleApiClient.ConnectionCallbacks {

    private GoogleApiClient mGoogleApiClient;
    private Messenger mMessenger;

    /** {@inheritDoc} */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mGoogleApiClient = new GoogleApiClient.Builder(this)
                            .addApi(Wearable.API)
                            .addConnectionCallbacks(this)
                            .build();

        mMessenger = new Messenger.Builder(mGoogleApiClient)
                                  .register(new StandardMessageReceiver(this))
                                  .register(new ConfirmMessageReceiver(this, PATH_SUCCESS_MESSAGE))
                                  .register(new ConfirmMessageReceiver(this, PATH_ERROR_MESSAGE))
                                  .register(new ResponseMessageReceiver())
                                  .rejectDecider(new Messenger.RejectDecider() {
                                        @Override
                                        public boolean shouldReject() {
                                            return false;
                                        }
                                    })
                                  .get();
    }

    /** {@inheritDoc} */
    @Override
    protected void onResume() {
        super.onResume();

        if (!mGoogleApiClient.isConnected()) {
            mGoogleApiClient.connect();
        }
    }

    /** {@inheritDoc} */
    @Override
    protected void onPause() {
        super.onPause();

        if (mGoogleApiClient.isConnected()) {
            mGoogleApiClient.disconnect();
            Wearable.MessageApi.removeListener(mGoogleApiClient, mMessenger);
        }
    }

    /** {@inheritDoc} */
    @Override
    public void onConnected(Bundle bundle) {
        Wearable.MessageApi.addListener(mGoogleApiClient, mMessenger);
    }

    /** {@inheritDoc} */
    @Override
    public void onConnectionSuspended(int i) {
        // TODO: Implement
    }
}

File: sample-wear/src/main/java/com/github/droibit/messenger/sample/MainActivity.kt
package com.github.droibit.messenger.sample

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.lifecycle.flowWithLifecycle
import androidx.lifecycle.lifecycleScope
import com.github.droibit.messenger.Messenger
import com.github.droibit.messenger.sample.model.ConfirmMessageHandler
import com.github.droibit.messenger.sample.model.ConfirmMessageHandler.Companion.PATH_ERROR_MESSAGE
import com.github.droibit.messenger.sample.model.ConfirmMessageHandler.Companion.PATH_SUCCESS_MESSAGE
import com.github.droibit.messenger.sample.model.ResponseMessageHandler
import com.github.droibit.messenger.sample.model.ResponseMessageHandler.Companion.PATH_REQUEST_MESSAGE
import com.github.droibit.messenger.sample.model.StandardMessageHandler
import com.github.droibit.messenger.sample.model.StandardMessageHandler.Companion.PATH_DEFAULT_MESSAGE
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.flow.collect
import kotlinx.coroutines.launch

@ExperimentalCoroutinesApi
class MainActivity : ComponentActivity(R.layout.activity_main) {

  private val messenger: Messenger by lazy {
    Messenger.Builder(this)
      .getNodesTimeout(2_000L)
      .sendMessageTimeout(2_500L)
      .build()
  }

  private val handlers = hashMapOf(
    PATH_DEFAULT_MESSAGE to StandardMessageHandler(this),
    PATH_SUCCESS_MESSAGE to ConfirmMessageHandler(this, PATH_SUCCESS_MESSAGE),
    PATH_ERROR_MESSAGE to ConfirmMessageHandler(this, PATH_ERROR_MESSAGE),
    PATH_REQUEST_MESSAGE to ResponseMessageHandler()
  )

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    lifecycleScope.launch {
      messenger.messageEvents
        .flowWithLifecycle(lifecycle)
        .collect {
          val handler = handlers.getValue(it.path)
          handler.onMessageReceived(messenger, it)
        }
    }
  }
}
